pkg/scheduler/schedule_one.go

```diff
@@ -309,12 +309,21 @@ func (sched *Scheduler) bindingCycle(
 		return status
 	}
 
-	// Any failures after this point cannot lead to the Pod being considered unschedulable.
+	// Any failures after this point cannot lead to the Pod being considered unschedulable and so we create the PodScheduled condition.
 	// We define the Pod as "unschedulable" only when Pods are rejected at specific extension points, and PreBind is the last one in the scheduling/binding cycle.
 	//
 	// We can call Done() here because
 	// we can free the cluster events stored in the scheduling queue sonner, which is worth for busy clusters memory consumption wise.
 	sched.SchedulingQueue.Done(assumedPod.UID)
+	podScheduled := &v1.PodCondition{
+		Type:   v1.PodScheduled,
+		Status: v1.ConditionTrue,
+	}
+	newStatus := assumedPod.Status.DeepCopy()
+	podutil.UpdatePodCondition(newStatus, podScheduled)
+	if err := util.PatchPodStatus(ctx, sched.client, assumedPod, newStatus); err != nil {
+		logger.Error(err, "Error updating pod with PodScheduled condition", "pod", klog.KObj(assumedPod))
+	}
 
 	// Run "bind" plugins.
 	if status := sched.bind(ctx, fwk, assumedPod, scheduleResult.SuggestedHost, state); !status.IsSuccess() {
```

Comment:
Seems `assumedPod`'s status is still the old one because https://github.com/kubernetes/kubernetes/blob/1504f10e7946f95a8b1da35e28e4c7453ff62775/pkg/scheduler/util/utils.go#L105-L134 didn't modify `old.Status` (or it would modify?)

I am not sure if `assumePod` keep the old status match your expecation at here.